torch/_inductor/codecache.py

```diff
@@ -418,24 +418,30 @@ def _reduce_fake_tensor(t):
 def _reduce_tensor(t):
     """
     See FxGraphCachePickler. Custom reducer to pickle Tensors.
+    If we see tensors, we know they're constants stored as attributes on
+    the GraphModule. Include the values in the key calculation. Small
+    tensors will be inlined, so we can't serve the same cache entry for
+    different constant values. Large constants are treated as parameters,
+    so we could conceivably reuse a cache entry. To do that, however,
+    PyCodeCache would need more complexity to create a new module from its
+    cache, but with the right constants attached as attributes.
     """
     if t.is_mkldnn:
         # TODO: These tensors don't currently pickle, so we can't cache a
         # compiled graph containing them. Just fail now. If mkldnn tensors
         # get pickling support, we can remove this.
         raise BypassFxGraphCache()
 
-    # If we see tensors, we know they're constants stored as attributes on
-    # the GraphModule. See tensor lowering; small constants are inlined. If
-    # we see a small tensor, therefore, no reference will ultimately remain
-    # in the generated code. So we need to include its value in the cache key.
-    # Large constants are effectively treated as inputs and we consider only
-    # their metadata.
+    # Very large tensors could be expensive to copy to cpu and hash; maybe
+    # not worth it. Bypass the cache at something arbitrarily large.
+    if t.numel() > 100000:
```

Comment:
On 2nd thought this looks gross. Should we just let it be slow and see if it's an actual problem?

Reply:
Why don't we time the pickling here, and if it's above some threshold warn and say to file an issue ? We have similar code for compiling triton kernels